cluster-autoscaler/core/scaledown/eligibility/eligibility_test.go

```diff
@@ -50,44 +50,66 @@ func TestFilterOutUnremovable(t *testing.T) {
 	noScaleDownNode.Annotations = map[string]string{ScaleDownDisabledKey: "true"}
 	SetNodeReadyState(noScaleDownNode, true, time.Time{})
 
+	unreadyNode := BuildTestNode("unready", 1000, 10)
+	SetNodeReadyState(unreadyNode, false, time.Time{})
+
 	bigPod := BuildTestPod("bigPod", 600, 0)
 	bigPod.Spec.NodeName = "regular"
 
 	smallPod := BuildTestPod("smallPod", 100, 0)
 	smallPod.Spec.NodeName = "regular"
 
 	testCases := []struct {
-		desc  string
-		nodes []*apiv1.Node
-		pods  []*apiv1.Pod
-		want  []string
+		desc             string
+		nodes            []*apiv1.Node
+		pods             []*apiv1.Pod
+		want             []string
+		scaleDownUnready bool
 	}{
 		{
-			desc:  "regular node stays",
-			nodes: []*apiv1.Node{regularNode},
-			want:  []string{"regular"},
+			desc:             "regular node stays",
+			nodes:            []*apiv1.Node{regularNode},
+			want:             []string{"regular"},
+			scaleDownUnready: true,
+		},
+		{
+			desc:             "recently deleted node is filtered out",
+			nodes:            []*apiv1.Node{regularNode, justDeletedNode},
+			want:             []string{"regular"},
+			scaleDownUnready: true,
+		},
+		{
+			desc:             "marked no scale down is filtered out",
+			nodes:            []*apiv1.Node{noScaleDownNode, regularNode},
+			want:             []string{"regular"},
+			scaleDownUnready: true,
 		},
 		{
-			desc:  "recently deleted node is filtered out",
-			nodes: []*apiv1.Node{regularNode, justDeletedNode},
-			want:  []string{"regular"},
+			desc:             "highly utilized node is filtered out",
+			nodes:            []*apiv1.Node{regularNode},
+			pods:             []*apiv1.Pod{bigPod},
+			want:             []string{},
+			scaleDownUnready: true,
 		},
 		{
-			desc:  "marked no scale down is filtered out",
-			nodes: []*apiv1.Node{noScaleDownNode, regularNode},
-			want:  []string{"regular"},
+			desc:             "underutilized node stays",
+			nodes:            []*apiv1.Node{regularNode},
+			pods:             []*apiv1.Pod{smallPod},
+			want:             []string{"regular"},
+			scaleDownUnready: true,
 		},
 		{
-			desc:  "highly utilized node is filtered out",
-			nodes: []*apiv1.Node{regularNode},
-			pods:  []*apiv1.Pod{bigPod},
-			want:  []string{},
+			desc:             "unready node stays",
+			nodes:            []*apiv1.Node{unreadyNode},
+			want:             []string{"unready"},
+			scaleDownUnready: true,
 		},
 		{
-			desc:  "underutilized node stays",
-			nodes: []*apiv1.Node{regularNode},
-			pods:  []*apiv1.Pod{smallPod},
-			want:  []string{"regular"},
+			// Ensure that unready node is filtered out when scale-down for unready is disabled
```

Comment:
Isn't this comment redundant with test case `desc`?

Reply:
Totally, I will remove it.